coins/src/adapters/moneyMarkets/morpho.ts

```diff
@@ -0,0 +1,277 @@
+        totalBorrowAssets: number;
+      };
+    } = {};
+    marketDataArray.forEach((m: any, i: number) => {
+      marketData[uniqueMarkets[i]] = m;
+    });
+
+    return marketData;
+  }
+
+  const currentTotalWithdrawables = aggregateWithdrawable(
+    currentVaultDatas,
+    currentMarketData
+  );
+  const previousTotalWithdrawables = aggregateWithdrawable(
+    previousVaultDatas,
+    previousMarketData
+  );
+
+  function aggregateWithdrawable(vaultDatas: VaultDatas, marketData: any) {
+    let totalWithdrawables: { [vault: string]: number } = {};
+    Object.keys(vaultDatas).map((vault: string) => {
+      totalWithdrawables[vault] = 0;
+      const { positions, markets } = vaultDatas[vault];
+
+      markets.map((market: string, i: number) => {
+        const { totalSupplyAssets, totalSupplyShares, totalBorrowAssets } =
+          marketData[market];
+        if (positions[i].supplyShares == 0) return;
+        const supplyAssets =
+          (positions[i].supplyShares * totalSupplyAssets) / totalSupplyShares;
+
+        const availableLiquidity = Math.max(
+          totalSupplyAssets - totalBorrowAssets,
+          0
+        );
+
+        const withdrawable = Math.min(supplyAssets, availableLiquidity);
+
+        totalWithdrawables[vault] += Number(withdrawable);
+      });
+    });
+
+    return totalWithdrawables;
+  }
+
+  const problemVaultList: string[] = [];
+  Object.keys(currentVaultDatas).map((vault: string) => {
+    const { totalAssets } = currentVaultDatas[vault];
+    if (totalAssets == 0) return;
+
+    const currentWithdrawable = currentTotalWithdrawables[vault];
+    const previousWithdrawable = previousTotalWithdrawables[vault];
+    if (currentWithdrawable / totalAssets > 0.01) return;
+
+    if (!previousVaultDatas[vault]) {
+      if (currentWithdrawable / totalAssets < 0.1)
+        console.log(
+          `${vault}: ${((currentWithdrawable / totalAssets) * 100).toFixed(2)}%`
+        );
+
+      problemVaultList.push(vault);
+      return;
+    }
+
+    const { totalAssets: previousTotalAssets } = previousVaultDatas[vault];
+    if (
+      previousWithdrawable &&
+      previousWithdrawable / previousTotalAssets > 0.95
+    )
+      return;
+
+    problemVaultList.push(vault);
+    console.log(
+      `${vault}: ${((currentWithdrawable / totalAssets) * 100).toFixed(2)}%`
+    );
+  });
+
+  const metadata = await getTokenInfoMap(api.chain, problemVaultList);
+
+  const writes: Write[] = [];
+  problemVaultList.map(async (vault: string) => {
+    const { symbol, decimals } = metadata[vault];
+    if (!symbol || !decimals) return;
+    addToDBWritesList(
+      writes,
+      api.chain,
+      vault,
+      0,
+      decimals,
+      symbol,
+      timestamp,
+      "morpho",
+      1.01
+    );
+  });
+
+  return writes;
+}
+
+async function getListaVaults(chain: string) {
+  const {
+    data: { list: vaults },
+  } = await fetch(listaConfig[chain].vaultInfo).then((r) => r.json());
```

Comment:
use cached Fetch so the price adapter will continue to work even if api gets changed